cfgmgr/coppmgr.cpp

```diff
@@ -0,0 +1,745 @@
+            }
+            else
+            {
+                trap_ids += list_item_delimiter + it.first;
+            }
+        }
+    }
+}
+
+void CoppMgr::doCoppTrapTask(Consumer &consumer)
+{
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key = kfvKey(t);
+        string op = kfvOp(t);
+        vector<FieldValueTuple> fvs;
+        string trap_ids;
+        string trap_group;
+        bool   conf_present = false;
+
+        if (op == SET_COMMAND)
+        {
+            /*Create case*/
+            if (m_coppTrapConfMap.find(key) != m_coppTrapConfMap.end())
+            {
+                trap_ids = m_coppTrapConfMap[key].trap_ids;
+                trap_group = m_coppTrapConfMap[key].trap_group;
+                conf_present = true;
+            }
+
+            bool null_cfg = false;
+            for (auto i: kfvFieldsValues(t))
+            {
+                if (fvField(i) == COPP_TRAP_GROUP_FIELD)
+                {
+                    trap_group = fvValue(i);
+                }
+                else if (fvField(i) == COPP_TRAP_ID_LIST_FIELD)
+                {
+                    trap_ids = fvValue(i);
+                }
+                else if (fvField(i) == "NULL")
+                {
+                    null_cfg = true;
+                }
+            }
+            if (null_cfg)
+            {
+                if (!m_coppTrapConfMap[key].trap_group.empty())
+                {
+                    SWSS_LOG_DEBUG("Deleting trap key %s", key.c_str());
+                    removeTrapIdsFromTrapGroup(m_coppTrapConfMap[key].trap_group,
+                            m_coppTrapConfMap[key].trap_ids);
+                    trap_ids.clear();
+                    setCoppTrapStateOk(key);
+                    getTrapGroupTrapIds(m_coppTrapConfMap[key].trap_group, trap_ids);
+                    fvs.clear();
+                    FieldValueTuple fv(COPP_TRAP_ID_LIST_FIELD, trap_ids);
+                    fvs.push_back(fv);
+                    if (!checkTrapGroupPending(m_coppTrapConfMap[key].trap_group))
+                    {
+                        m_appCoppTable.set(m_coppTrapConfMap[key].trap_group, fvs);
+                    }
+                    setCoppGroupStateOk(m_coppTrapConfMap[key].trap_group);
+                    m_coppTrapConfMap[key].trap_group = "";
+                    m_coppTrapConfMap[key].trap_ids = "";
+                }
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+            /*Duplicate check*/
+            if (conf_present &&
+                (trap_group == m_coppTrapConfMap[key].trap_group) &&
+                (trap_ids == m_coppTrapConfMap[key].trap_ids))
+            {
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+            removeTrapIdsFromTrapGroup(m_coppTrapConfMap[key].trap_group,
```

Comment:
Add a comment for explaining why the trapid is removed

Reply:
Done